Update to be Rails 5.1 compatible

runningferret opened the pull request:
Cribbed from https://github.com/pelle/oauth-plugin/pull/156. I had to manually tweak the .patch to get it to apply due to some code we apparently commented out.

I'm creating this PR against `kenna-tweaks` but have not decided what the actual right path forward is.

runningferret committed:
Update to be Rails 5.1 compatible

Cribbed from https://github.com/pelle/oauth-plugin/pull/156. I had to manually tweak the .patch to get it to apply due to some code we apparently commented out.

bschmeck reviewed (commented):
This appears to be the correct change to address the `before_filter` usage.  I just noticed in Kibana that we're also seeing deprecation warnings for the use of `render text:` in the controller:

```
DEPRECATION WARNING: `render :text` is deprecated because it does not actually render a `text/plain` response. Switch to `render plain: 'plain text'` to render as `text/plain`, `render html: '<strong>HTML</strong>'` to render as `text/html`, or `render body: 'raw'` to match the deprecated behavior and render with the default Content-Type, which is `text/html`. (called from access_token at /opt/apps/source/conduit/releases/20191108201601Z/vendor/cache/oauth-plugin-c5ec12bde52e/lib/oauth/controllers/provider_controller.rb:29)```

runningferret committed:
Update render :text to :plain

Rails 5 deprecates render :text (as it merely renders the text but does not modify the Content-Type header) and instead specifies render :plain for text/plain.

mstruve commented:
I had the biggest, "WTF is going on moment?!" when I got a notification for this PR, but alas it makes sense bc its actually public.

runningferret committed:
Merge remote-tracking branch 'origin/master' into rails_5_1_update

rudyrigot commented:
And I had the biggest, "WTF is going on moment?!" when I got a notification from @mstruve for this PR. "Wait, is she back??" ;)
